src/main/java/org/cactoos/iterator/Paged.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.cactoos.Func;
+import org.cactoos.Scalar;
+import org.cactoos.func.UncheckedFunc;
+import org.cactoos.scalar.Sticky;
+import org.cactoos.scalar.Unchecked;
+
+/**
+ * Paged iterator.
+ * Elements will continue to be provided so long as {@code next} produces
+ * non-empty iterators.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @param <I> Subtype of item iterator
+ * @param <X> Type of item
+ * @since 0.49
+ */
+public final class Paged<I extends Iterator<X>, X> implements Iterator<X> {
```

Comment:
@baudoliver7 why do we need this `I` here, can't we just use `Iterator<X>` directly in the class without introducing `I`? If it's doable, let's get rid of it :) Or instead just add a todo for creating a new task for it. I think we should try to get rid of it here and also in iterable.Paged, maybe using `Iterator<? extends X>` for example, because we don't have to constrain the `Func` to give exactly the same Iterator type as long as it  is an iterator that provides `X`.

Reply:
@victornoel I agree with you. I do it now.

Reply:
@victornoel After having thought, I found out it is not doable to get rid of it unless we want to replace `I` everywhere it is used by `? extends Iterator<X>`, which is not easy to read. It's for the sake of clarity that `I` has been introduced.

Reply:
@baudoliver7 the problem is that it is incorrect like this because it means we cannot produce different subclasses of iterator with this, only one that matches the `I`. Clarity is not important here, what is important is that the class is usable.

I propose the following: let's add a todo to get rid of this `I` so that we can now focus on the rest of the PR :)

Reply:
@victornoel Ok

Reply:
@victornoel Done.